### PR TITLE
Implement sendmail

### PR DESCRIPTION
**Pull request by couling**
Implemented `sendmail` as an alternative to SMTP.
A number of gogs users have requested the ability to use `sendmail` as an alternative to SMTP: https://github.com/gogits/gogs/issues/997
For some users setting up SMTP accounts for gogs is a lot more trouble than it's worth.

To keep this change simple I've not added any addational configuration parameter.  Instead I've reused the SMTP host.  If this contains a "/" then it will be interpreted as the location of the sendmail binary.  This is safe because hostnames can't contain slashes. 

Other fields such as username and password will be ignored for sendmail.

### CODE REVIEW

**Comment by tboerger** [comment]
I prefer real config options for that.

**Comment by tboerger** [comment]
Beside that the linting fails within Drone.

**Comment by couling** [comment]
Okay. I'll take a look and re-submit

File: modules/mailer/mailer.go [hunk 1]
```diff
@@ -195,6 +195,47 @@ func (s *Sender) Send(from string, to []string, msg io.WriterTo) error {
 	return client.Quit()
 }
 
+func (s *Sender) sendSendmail(from string, to []string, msg io.WriterTo) error {
```
Comment (bkcsoft) [inline modules/mailer/mailer.go]: This should really be implemented as a separate `Sender` instead, this way you don't have to do checks for every single sent email...

Config-reading and setup could be done here: https://github.com/go-gitea/gitea/pull/355/files#diff-bdf9b0a2ed1413ee986db72787e5b5d7R265

**Review by bkcsoft — CHANGES_REQUESTED** [review]

**Comment by couling** [comment]
Okay. These changes should have addressed previous concerns.

**Review by bkcsoft — APPROVED** [review]

**Comment by lunny** [comment]
LGTM

**Comment by bkcsoft** [comment]
LGTM

**Comment by lunny** [comment]
And maybe we need add some docs for all the new features after it is merged.

**Comment by couling** [comment]
@lunny What would the best place to document this be?

**Comment by lunny** [comment]
send PR to github.com/go-gitea/docs

**Pull request merged**